asset-transfer-basic/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,235 @@
+            );
+
+            // Submit a transaction, blocking until the transaction has been committed on the ledger
+            await contract.submitTransaction('InitLedger');
+            console.log('*** Result: committed');
+
+            console.log(
+                '\n--> Evaluate Transaction: GetAllAssets, function returns all the current assets on the ledger',
+            );
+
+            let result = await contract.evaluateTransaction('GetAllAssets');
+            console.log(`*** Result: ${Buffer.from(result).toString()}`);
+
+            console.log(
+                '\n--> Submit Transaction: CreateAsset, creates new asset with ID, color, owner, size, and appraisedValue arguments',
+            );
+
+            await contract.submitTransaction(
+                'CreateAsset',
+                'asset13',
+                'yellow',
+                '5',
+                'Tom',
+                '1300',
+            );
+            console.log('*** Result: committed');
+
+            console.log(
+                '\n--> Evaluate Transaction: ReadAsset, function returns an asset with a given assetID',
+            );
+            result = await contract.evaluateTransaction('ReadAsset', 'asset13');
+
+            console.log(`*** Result: ${Buffer.from(result).toString()}`);
+
+            console.log(
+                '\n--> Evaluate Transaction: AssetExists, function returns "true" if an asset with given assetID exist',
+            );
+            result = await contract.evaluateTransaction('AssetExists', 'asset1');
+            console.log(`*** Result: ${Buffer.from(result).toString()}`);
+
+            console.log(
+                '\n--> Submit Transaction: UpdateAsset asset1, change the appraisedValue to 350',
+            );
+
+            await contract.submitTransaction('UpdateAsset',
+                'asset1', 'blue', '5', 'Tomoko', '350',
+            );
+            console.log('*** Result: committed');
+
+            console.log(
+                '\n--> Submit Transaction: UpdateAsset asset1 asynchronously, change the appraisedValue to 400',
+            );
+
+            // Submit transaction asynchronously, blocking until the transaction has been sent to the orderer, and allowing
+            // this thread to process the chaincode response (e.g. update a UI) without waiting for the commit notification
+
+            const commit = await contract.submitAsync('UpdateAsset', {
+                arguments: ['asset1', 'blue', '5', 'Tomoko', '400'],
+            });
+
+            console.log('*** Waiting for transaction commit');
+
+            const successful = await commit.isSuccessful();
+            if (!successful) {
+                const status = await commit.getStatus();
+                throw new Error(`Transaction ${commit.getTransactionId()} failed to commit with status code: ${status}`);
+            }
```

Comment:
This flow has just changed. Sorry! It would be worth updating to the latest published version of _fabric-gateway_ and reworking to use the current flow, which is something like:
```typescript
const status = await commit.getStatus();
if (!status.successful) {
    throw new Error(`Transaction ${status.transactionId} failed to commit with status code ${status.code}`);
}
```